File: SolveEquation/c#/exeWF/Form1.cs
using System;
using System.Windows.Forms;

namespace SolveEquation
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void OnTextChanged(object sender, EventArgs e)
        {
            double[] z = new double[10];
            double.TryParse(txtR0.Text, out z[0]);
            double.TryParse(txtI0.Text, out z[1]);
            double.TryParse(txtR1.Text, out z[2]);
            double.TryParse(txtI1.Text, out z[3]);
            double.TryParse(txtR2.Text, out z[4]);
            double.TryParse(txtI2.Text, out z[5]);
            double.TryParse(txtR3.Text, out z[6]);
            double.TryParse(txtI3.Text, out z[7]);
            double.TryParse(txtR4.Text, out z[8]);
            double.TryParse(txtI4.Text, out z[9]);
#if false
            double[] x = SolveEquationNET.Equation.Solve(z);
#else
            double[] x = Solve.SolveEquation(z);
#endif
            int n = 0;
            if (x != null)
            {
                n   =   x.Length / 3;
            }
            const double d = 1e-10;
            if (n >= 1)
            {
                txtRA.Text = x[0].ToString();
                txtIA.Text = x[1].ToString();
                lblA.Text = x[2].ToString("0.e+000");
                lblA.ForeColor = Math.Abs(x[2]) > d ? System.Drawing.Color.Red : System.Drawing.Color.Blue;
            }
            else
            {
                lblA.Text = ""; txtRA.Text = ""; txtIA.Text = "";
            }
            if (n >= 2)
            {
                txtRB.Text = x[3].ToString();
                txtIB.Text = x[4].ToString();
                lblB.Text = x[5].ToString("0.e+000");
                lblB.ForeColor = Math.Abs(x[5]) > d ? System.Drawing.Color.Red : System.Drawing.Color.Blue;
            }
            else
            {
                lblB.Text = ""; txtRB.Text = ""; txtIB.Text = "";
            }
            if (n >= 3)
            {
                txtRC.Text = x[6].ToString();
                txtIC.Text = x[7].ToString();
                lblC.Text = x[8].ToString("0.e+000");
                lblC.ForeColor = Math.Abs(x[8]) > d ? System.Drawing.Color.Red : System.Drawing.Color.Blue;
            }
            else
            {
                lblC.Text = ""; txtRC.Text = ""; txtIC.Text = "";
            }
            if (n >= 4)
            {
                txtRD.Text = x[9].ToString();
                txtID.Text = x[10].ToString();
                lblD.Text = x[11].ToString("0.e+000");
                lblD.ForeColor = Math.Abs(x[11]) > d ? System.Drawing.Color.Red : System.Drawing.Color.Blue;
            }
            else
            {
                lblD.Text = ""; txtRD.Text = ""; txtID.Text = "";
            }
        }
    }
}

File: SolveEquation/vb/vb2010/Form1.vb
Option Strict Off
Option Explicit On
Friend Class Form1
	Inherits System.Windows.Forms.Form
	
    Private Declare Function SolveEquation Lib "SolveEquationDll.dll" (ByRef z As Double, ByRef x As Double) As Integer
	
    Private Sub txtR0_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtR0.TextChanged
        Call OnTextChange()
    End Sub
	
    Private Sub txtI0_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtI0.TextChanged
        Call OnTextChange()
    End Sub
	
    Private Sub txtR1_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtR1.TextChanged
        Call OnTextChange()
    End Sub
	
    Private Sub txtI1_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtI1.TextChanged
        Call OnTextChange()
    End Sub
	
    Private Sub txtR2_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtR2.TextChanged
        Call OnTextChange()
    End Sub
	
    Private Sub txtI2_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtI2.TextChanged
        Call OnTextChange()
    End Sub
	
    Private Sub txtR3_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtR3.TextChanged
        Call OnTextChange()
    End Sub
	
    Private Sub txtI3_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtI3.TextChanged
        Call OnTextChange()
    End Sub
	
    Private Sub txtR4_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtR4.TextChanged
        Call OnTextChange()
    End Sub
	
    Private Sub txtI4_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtI4.TextChanged
        Call OnTextChange()
    End Sub
	
	Private Sub OnTextChange()
		Dim z(10) As Double
		On Error Resume Next
		z(0) = CDbl(txtR0.Text)
		z(1) = CDbl(txtI0.Text)
		z(2) = CDbl(txtR1.Text)
		z(3) = CDbl(txtI1.Text)
		z(4) = CDbl(txtR2.Text)
		z(5) = CDbl(txtI2.Text)
		z(6) = CDbl(txtR3.Text)
		z(7) = CDbl(txtI3.Text)
		z(8) = CDbl(txtR4.Text)
		z(9) = CDbl(txtI4.Text)
		Dim x(12) As Double
		Dim n As Integer '有效解的个数
        n = SolveEquation(z(0), x(0)) '解方程
		Dim d As Double
		d = 0.0000000001
		If n >= 1 Then
			txtRA.Text = CStr(x(0))
			txtIA.Text = CStr(x(1))
            lblA.Text = x(2).ToString("0.e+000")
            If System.Math.Abs(x(2)) > d Then
                lblA.ForeColor = System.Drawing.Color.Red
            Else
                lblA.ForeColor = System.Drawing.Color.Blue
            End If
		Else
			txtRA.Text = ""
			txtIA.Text = ""
            lblA.Text = ""
		End If
		If n >= 2 Then
			txtRB.Text = CStr(x(3))
			txtIB.Text = CStr(x(4))
            lblB.Text = x(5).ToString("0.e+000")
			If System.Math.Abs(x(5)) > d Then
				lblB.ForeColor = System.Drawing.Color.Red
			Else
				lblB.ForeColor = System.Drawing.Color.Blue
			End If
		Else
			txtRB.Text = ""
			txtIB.Text = ""
            lblB.Text = ""
        End If
		If n >= 3 Then
			txtRC.Text = CStr(x(6))
			txtIC.Text = CStr(x(7))
            lblC.Text = x(8).ToString("0.e+000")
			If System.Math.Abs(x(8)) > d Then
				lblC.ForeColor = System.Drawing.Color.Red
			Else
				lblC.ForeColor = System.Drawing.Color.Blue
			End If
		Else
			txtRC.Text = ""
			txtIC.Text = ""
            lblC.Text = ""
        End If
		If n >= 4 Then
			txtRD.Text = CStr(x(9))
			txtID.Text = CStr(x(10))
            lblD.Text = x(11).ToString("0.e+000")
			If System.Math.Abs(x(11)) > d Then
				lblD.ForeColor = System.Drawing.Color.Red
			Else
				lblD.ForeColor = System.Drawing.Color.Blue
			End If
		Else
			txtRD.Text = ""
			txtID.Text = ""
            lblD.Text = ""
        End If
	End Sub
End Class